analyzers/src/SonarAnalyzer.Common/Rules/PropertyGetterWithThrowBase.cs

```diff
@@ -68,7 +68,8 @@ protected sealed override void Initialize(SonarAnalysisContext context)
 
                             var symbol = c.SemanticModel.GetSymbolInfo(throwExpression).Symbol;
                             if (symbol == null ||
-                                symbol.ContainingType.DerivesFromAny(AllowedExceptionBaseTypes))
+                                symbol.ContainingType.DerivesFromAny(AllowedExceptionBaseTypes) ||
+                                (symbol as IMethodSymbol)?.ReturnType?.DerivesFromAny(AllowedExceptionBaseTypes) == true)
```

Comment:
From my understanding, we should not raise _at all_ if the throw type is one of the allowed ones, no matter how ithe exception is generated.

To get two birds with one stone, you can check the type symbol of the expression instead of trying to find the containing symbol.

Something like this:

```csharp
var type = c.SemanticModel.GetTypeInfo(throwExpression).Type;
if (type is null || type.DerivesFromAny(AllowedExceptionBaseTypes))
{
    return;
}

```

Reply:
That's much cleaner. Thank you.